### PR TITLE
Ability to exclude paths in input-output-logger

### PR DESCRIPTION
**Pull request by roni-frantchi**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Adds the ability to exclude paths from input/output logger.  
Very useful to reduce log verbosity and scrub-off secrets or sensitive data.  

Where has this been tested?
---------------------------
**Node.js Versions:** v12.16.1

**Middy Versions:** 1.0.0-beta

**AWS SDK Versions:** 2.556.0

Todo list
---------

- [x] Feature/Fix fully implemented
- [x] Added tests
- [x] Updated relevant documentation
- [x] Updated relevant examples

### CODE REVIEW

File: packages/input-output-logger/index.js [hunk 1]
```diff
@@ -1,21 +1,29 @@
+const omit = require('lodash/omit')
+
 module.exports = (opts) => {
   const defaults = {
-    logger: data => console.log(JSON.stringify(data, null, 2))
+    logger: data => console.log(JSON.stringify(data, null, 2)),
+    exclude: []
```
Comment (lmammino) [inline packages/input-output-logger/index.js:6]: Maybe a little bit more explicit to call this something like `omitPaths`?

File: packages/input-output-logger/README.md [hunk 2]
```diff
@@ -45,6 +45,7 @@ npm install --save @middy/input-output-logger
 ## Options
 
 `logger` property accept a function (default `console.log`)
+`exclude` property accept an array of paths to exclude from the log (default `[]`)
```
Comment (lmammino) [inline packages/input-output-logger/README.md:48]: I initially had a feeling that this would exclude the entire log message if an "exclude" path would match it.

Going through the code it seems that this works a bit more as a "field redaction" mechanism (e.g. to remove passwords from logs).

If that's the case, maybe we can be a bit more explicit here with something like this:

>  `exclude` property accepts an array of paths that will be used to remove particular fields from the logged objects. This could serve as a simple way to redact sensitive data from logs (default `[]`).

I was also suggesting that changing `exclude` to something more explicit can help to clarify the scope here. What do you think?

File: packages/input-output-logger/__tests__/index.js [hunk 3]
```diff
@@ -18,4 +18,19 @@ describe('📦 Middleware Input Output Logger', () => {
     expect(logger).toHaveBeenCalledWith({ event: { foo: 'bar', fuu: 'baz' } })
     expect(logger).toHaveBeenCalledWith({ response: { message: 'hello world' } })
   })
+  test('It should exclude paths', async () => {
+    const logger = jest.fn()
+
+    const handler = middy((event, context, cb) => {
+      cb(null, { message: 'hello world', bar: 'bi' })
```
Comment (lmammino) [inline packages/input-output-logger/__tests__/index.js:25]: what happens if `response` is not an object (e.g. a string). Should we add an extra test to check for these edge cases?

File: packages/input-output-logger/index.js [hunk 4]
```diff
@@ -1,21 +1,29 @@
...
   }
 
-  const options = Object.assign({}, defaults, opts)
+  const { logger, exclude } = Object.assign({}, defaults, opts)
+
+  const omitAndLog = data => {
+    const message = omit(data, exclude)
```
Comment (lmammino) [inline packages/input-output-logger/index.js:12]: maybe for clarity let's rename this to `redactedMessage`

File: packages/input-output-logger/package.json [hunk 5]
```diff
@@ -44,10 +44,12 @@
     "aws-sdk": ">=2.221.1"
   },
   "dependencies": {
-    "@types/node": "^10.0.8"
+    "lodash": "^4.17.15"
   },
   "devDependencies": {
     "@middy/core": "^1.0.0-beta.10",
+    "@types/node": "^10.0.8",
```
Comment (lmammino) [inline packages/input-output-logger/package.json:51]: good catch here!

File: packages/input-output-logger/index.js (hunk 1 above)
Comment (roni-frantchi) [reply packages/input-output-logger/index.js:6]: Done [here](https://github.com/middyjs/middy/pull/507/commits/cd02db468a56d2184577eaeae40e5fadb0468508)

File: packages/input-output-logger/index.js (hunk 4 above)
Comment (roni-frantchi) [reply packages/input-output-logger/index.js]: Done [here](https://github.com/middyjs/middy/pull/507/commits/cd02db468a56d2184577eaeae40e5fadb0468508)

File: packages/input-output-logger/README.md (hunk 2 above)
Comment (roni-frantchi) [reply packages/input-output-logger/README.md]: Done [here](https://github.com/middyjs/middy/pull/507/commits/b213c25704ce52b3958a9ca8c23a389394952bdc)

File: packages/input-output-logger/__tests__/index.js (hunk 3 above)
Comment (roni-frantchi) [reply packages/input-output-logger/__tests__/index.js]: Done [here](https://github.com/middyjs/middy/pull/507/commits/d3b0fc04503b529ee1929ef6cb2e5bcf58948a7c)

File: packages/input-output-logger/index.js (hunk 4 above)
Comment (lmammino) [reply packages/input-output-logger/index.js]: Tnx

File: packages/input-output-logger/__tests__/index.js (hunk 3 above)
Comment (lmammino) [reply packages/input-output-logger/__tests__/index.js]: Tnx

File: packages/input-output-logger/README.md (hunk 2 above)
Comment (lmammino) [reply packages/input-output-logger/README.md]: Awesome, thanks!

File: packages/input-output-logger/index.js (hunk 1 above)
Comment (lmammino) [reply packages/input-output-logger/index.js]: ;)

**Comment by lmammino** [comment]
Releasing this in [`1.0.0-beta.11`](https://github.com/middyjs/middy/releases/tag/1.0.0-beta.11)

**Comment by roni-frantchi** [comment]
Thanks @lmammino! ❤️
